README.md

```diff
@@ -1 +1,45 @@
-# dataverse-frontend
\ No newline at end of file
+# Dataverse Frontend
+
+[![Tests](https://github.com/IQSS/dataverse-frontend/actions/workflows/test.yml/badge.svg)](https://github.com/IQSS/dataverse-frontend/actions/workflows/test.yml)
+
+## Getting Started
+
+First install node >=16 and npm >=8. Recommended versions `node v19` and `npm v9`.
+
+### `npm install`
+
+Installs dependencies.
+
+
+## Available Scripts
+
+In the project directory, you can run at any time:
+
+### `npm start`
+
+Runs the app in the development mode.\
+Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
+
+The page will reload when you make changes.\
+You may also see any lint errors in the console.
+
+### `npm test`
+
+Launches the test runner for the unit tests in the interactive watch mode.
+
+### `npm run cy:run`
+
+Launches the Cypress test runner for the end-to-end tests. \
+If you prefer to see the tests executing in cypress you can run `npm run cy:open`
+
+### `npm run lint`
+
+Launches the linter. To automatically fix the errors run `npm run lint:fix`
+
+### `npm run format`
```

Comment:
I have tested this command and some files in the repository have been formatted. It looks like you didn't run it for the last commit. Could you run it and commit the changes?

Perhaps it would be nice to run this command on a pre-commit hook.

Reply:
Can we run the formatter automatically with a post commit hook or whatever? That way, if people forget to run it, no problem! Git will run it for you.

Reply:
Good point!

ESlint also has the rules of prettier, then I was confident that the lint workflow was checking te formatting too, but ESlint doesn't run on .json,.yml or md files. At the same time I realized that IntelliJ is not running prettier on .json and .yml files, so even with formatting configured on save those files were excluded.

I've modified the scripts to run prettier on json, yml and m files so now the Github Workflow will fail if this type of files are not formatted.

About the pre-commit hook I added it too, thank you for your recommendation! Now is running the lint, format and unit test scripts. It is super fast and now it prevents commits not passing lint or unit tests. Maybe we should leave the GitHub actions for pull requests only, now that the pre-commit is checking lint and unit tests

Reply:
@pdurbin I wanted to do exactly that, running formatter and linter automatically and then commit the changes automatically. But doing the automatic commit is the hard part, you have to give .git writing permissions, so I don't see it really convenient

I also tried the [pre-commit](https://pre-commit.com) approach but the hooks are already provided and prettier hook is not the same as our prettier configuration :(

Finally I decided to use [pre-commit npm package ](https://www.npmjs.com/package/pre-commit) it doesn't commit the changes automatically but it does run the formatter and linter automatically on commit. Then you have to manually add those changes to your commit

Do you know of a better approach to automatically run and commit the formatting?

Reply:
Hmm. I don't but let me ping @poikilotherm because he worked hard on this for over at https://github.com/gdcc/xoai/pull/36

Reply:
Thank you Philip! I finally managed to add the formatting changes to the commit automatically. I just had to add the `git add .` command to the pre-commit scripts